### ISSUE TITLE
Completion popup don't appear before writing whole variable name

### ISSUE THREAD

**Issue by gbaranski** [helix-editor/helix#558]
### Reproduction steps
1. Open LSP supported file.
2. Create variable named `something`
3. Type in `something` slowly, and notice that nothing appears.

Helix:

https://user-images.githubusercontent.com/34353640/128600734-efb29b50-12f1-4493-ac27-61908a851746.mp4

Neovim:

https://user-images.githubusercontent.com/34353640/128600729-66eee4e4-801b-40d8-be01-d0268ee6e91c.mp4

VSCode:

https://user-images.githubusercontent.com/34353640/128600804-6ed400ab-0c00-47f2-b59d-edb2c16f45f8.mp4

### Environment

- Platform: Linux
- Helix version: v0.3.0-229-gba72934

**Comment by archseer** [comment]
We currently only support triggers specified by the LSP, in this case `.`. You can always manually trigger completion via `ctrl-x` but I also want to trigger on idle in the future. It requires an idle timeout hook, I started working on this but got distracted

**Comment by sudormrfbin** [comment]
Wouldn't complete-as-you-type offer a better experience than complete on idle ? It'll consume more resources I suppose, but I think we can make it configurable and optimize as necessary.

**Comment by Inoshy** [comment]
Hi, sorry for bothering you all. but how do you get these completions? I tried adding the sentence in my config: 
```
[lsp]
display-messages = true
```
But this gives me nothing. I have rls installed and running on vscode, but nothing pops up in helix.

**Comment by gbaranski** [comment]
> Hi, sorry for bothering you all. but how do you get these completions? I tried adding the sentence in my config:
> 
> ```
> [lsp]
> display-messages = true
> ```
> 
> But this gives me nothing. I have rls installed and running on vscode, but nothing pops up in helix.
> 
> Would you share your `config.toml`? it would be super helpful to me.

What languages you've tried?

**Comment by sudormrfbin** [comment]
@Inoshy You need rust-analyzer [installed separately](https://rust-analyzer.github.io/manual.html#rust-analyzer-language-server-binary) and placed in your `$PATH` for helix to use it.

**Comment by Inoshy** [comment]
@sudormrfbin thanks! I will try to do it and see what happens.

**Comment by Inoshy** [comment]
@sudormrfbin hi, sorry for disturbing again. I installed a package called `rust-analyzer-git` from the AUR, it installed the binary in `usr/bin`. I have run the `echo "$PATH"` command and it seems to be `usr/bin` is in the path.
I am launching rust-analyzer from terminal and when I open helix with a rust file it still isnt showing completion popups. what am I missing?

**Comment by Inoshy** [comment]
> > Hi, sorry for bothering you all. but how do you get these completions? I tried adding the sentence in my config:
> > ```
> > [lsp]
> > display-messages = true
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > But this gives me nothing. I have rls installed and running on vscode, but nothing pops up in helix.
> > Would you share your `config.toml`? it would be super helpful to me.
> 
> What languages you've tried?

Hi, sorry I did not notice the reply. I have only tried rust :)

**Comment by archseer** [comment]
Run `which rust-analyzer` to confirm it's in your $PATH. Also run helix with verbose logging `hx -v` and inspect `~/.cache/helix/helix.log` if there's any LSP activity. rust-analyzer also takes a long time to initially index all the code, so there might not be any completions or popups for the first 5-15s.

If you still have problems, please open a separate issue and we can troubleshoot and discuss there -- this one is specifically for completion behavior.

**Comment by archseer** [comment]
> Wouldn't complete-as-you-type offer a better experience than complete on idle ? It'll consume more resources I suppose, but I think we can make it configurable and optimize as necessary.

I think a low enough idle time is essentially as-you-type, and idle is easy to implement first. You'll notice it takes some time before the completion first pops up only after a few characters so there's a min check in there too. VSCode behavior uses quite a few tricks: https://github.com/roxma/nvim-completion-manager/issues/30#issuecomment-283281158
